IntegrationTests/tests_04_performance/test_01_resources/shared.swift

```diff
@@ -136,3 +136,119 @@ func withAutoReleasePool<T>(_ execute: () throws -> T) rethrows -> T {
     }
     #endif
 }
+
+// MARK: UDP Shared
+
+/// Repeatedly send, echo and receive a number of UDP requests.
+/// Based on the UDPEchoClient/Server example.
+enum UdpShared {
+    private final class EchoHandler: ChannelInboundHandler {
+        public typealias InboundIn = ByteBuffer
+        public typealias OutboundOut = ByteBuffer
+
+        public func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+            // As we are not really interested getting notified on success or failure we just pass nil as promise to
+            // reduce allocations.
+            context.write(data, promise: nil)
+        }
+
+        public func channelReadComplete(context: ChannelHandlerContext) {
+            // As we are not really interested getting notified on success or failure we just pass nil as promise to
+            // reduce allocations.
+            context.flush()
+        }
+
+        public func errorCaught(context: ChannelHandlerContext, error: Error) {
+            print("error: ", error)
+
+            // As we are not really interested getting notified on success or failure we just pass nil as promise to
+            // reduce allocations.
+            context.close(promise: nil)
+        }
+    }
+
+    private final class EchoHandlerClient: ChannelInboundHandler {
+        public typealias InboundIn = AddressedEnvelope<ByteBuffer>
+        public typealias OutboundOut = AddressedEnvelope<ByteBuffer>
+        private var repetitionsRemaining: Int
+        
+        private let remoteAddress: SocketAddress
+        
+        init(remoteAddress: SocketAddress,
+             numberOfRepetitions: Int) {
+            self.remoteAddress = remoteAddress
+            self.repetitionsRemaining = numberOfRepetitions
+        }
+        
+        public func channelActive(context: ChannelHandlerContext) {
+            // Channel is available. It's time to send the message to the server to initialize the ping-pong sequence.
+            sendSomeDataIfDesiredOrClose(context: context)
+        }
+        
+        private func sendSomeDataIfDesiredOrClose(context: ChannelHandlerContext) {
+            if (repetitionsRemaining > 0) {
+                repetitionsRemaining -= 1
+                
+                // Set the transmission data.
+                let line = "Something to send there and back again."
+                var buffer = context.channel.allocator.buffer(capacity: line.utf8.count)
+                buffer.writeString(line)
+                
+                // Forward the data.
+                let envolope = AddressedEnvelope<ByteBuffer>(remoteAddress: remoteAddress, data: buffer)
+                
+                context.writeAndFlush(self.wrapOutboundOut(envolope), promise: nil)
+            }
```

Comment:
nit: `else` fell off the `}`

Reply:
Will fix - Spent 25 years avoiding KnR bracketing.  Hate it :-(

Reply:
haha, this will be all over once the Swift community decided on the "official style". Then we can all type now we prefer and the formatter will make it "right".

Reply:
fixed.